File: source/cpBase/CPRequestBaseClass.vb
'
' documentation should be in a new project that inherits these classes. The class names should be the object names in the actual cp project
'
Namespace Contensive.BaseClasses
    Public MustInherit Class CPRequestBaseClass
        'Public Sub New(ByVal cmcObj As Contensive.Core.cpCoreClass, ByRef CPParent As CPBaseClass)
        Public MustOverride ReadOnly Property Browser() As String 'Implements BaseClasses.CPRequestBaseClass.Browser
        Public MustOverride ReadOnly Property BrowserIsIE() As Boolean 'Implements BaseClasses.CPRequestBaseClass.BrowserIsIE
        Public MustOverride ReadOnly Property BrowserIsMac() As Boolean 'Implements BaseClasses.CPRequestBaseClass.BrowserIsMac
        Public MustOverride ReadOnly Property BrowserIsMobile() As Boolean 'Implements BaseClasses.CPRequestBaseClass.BrowserIsMobile
        Public MustOverride ReadOnly Property BrowserIsWindows() As Boolean 'Implements BaseClasses.CPRequestBaseClass.BrowserIsWindows
        Public MustOverride ReadOnly Property BrowserVersion() As String 'Implements BaseClasses.CPRequestBaseClass.BrowserVersion
        Public MustOverride Function Cookie(ByVal CookieName As String) As String 'Implements BaseClasses.CPRequestBaseClass.Cookie
        Public MustOverride ReadOnly Property CookieString() As String 'Implements BaseClasses.CPRequestBaseClass.CookieString
        Public MustOverride ReadOnly Property Form() As String 'Implements BaseClasses.CPRequestBaseClass.Form
        Public MustOverride ReadOnly Property FormAction() As String 'Implements BaseClasses.CPRequestBaseClass.FormAction
        Public MustOverride Function GetBoolean(ByVal RequestName As String) As Boolean 'Implements BaseClasses.CPRequestBaseClass.GetBoolean
        Public MustOverride Function GetDate(ByVal RequestName As String) As Date 'Implements BaseClasses.CPRequestBaseClass.GetDate
        Public MustOverride Function GetInteger(ByVal RequestName As String) As Integer 'Implements BaseClasses.CPRequestBaseClass.GetInteger
        Public MustOverride Function GetNumber(ByVal RequestName As String) As Double 'Implements BaseClasses.CPRequestBaseClass.GetNumber
        Public MustOverride Function GetText(ByVal RequestName As String) As String 'Implements BaseClasses.CPRequestBaseClass.GetText
        Public MustOverride ReadOnly Property Host() As String 'Implements BaseClasses.CPRequestBaseClass.Host
        Public MustOverride ReadOnly Property HTTPAccept() As String 'Implements BaseClasses.CPRequestBaseClass.HTTPAccept
        Public MustOverride ReadOnly Property HTTPAcceptCharset() As String 'Implements BaseClasses.CPRequestBaseClass.HTTPAcceptCharset
        Public MustOverride ReadOnly Property HTTPProfile() As String 'Implements BaseClasses.CPRequestBaseClass.HTTPProfile
        Public MustOverride ReadOnly Property HTTPXWapProfile() As String 'Implements BaseClasses.CPRequestBaseClass.HTTPXWapProfile
        Public MustOverride ReadOnly Property Language() As String 'Implements BaseClasses.CPRequestBaseClass.Language
        Public MustOverride ReadOnly Property Link() As String 'Implements BaseClasses.CPRequestBaseClass.Link
        Public MustOverride ReadOnly Property LinkForwardSource() As String 'Implements BaseClasses.CPRequestBaseClass.LinkForwardSource
        Public MustOverride ReadOnly Property LinkSource() As String 'Implements BaseClasses.CPRequestBaseClass.LinkSource
        Public MustOverride ReadOnly Property Page() As String 'Implements BaseClasses.CPRequestBaseClass.Page
        Public MustOverride ReadOnly Property Path() As String 'Implements BaseClasses.CPRequestBaseClass.Path
        Public MustOverride ReadOnly Property PathPage() As String 'Implements BaseClasses.CPRequestBaseClass.PathPage
        Public MustOverride ReadOnly Property Protocol() As String 'Implements BaseClasses.CPRequestBaseClass.Protocol
        Public MustOverride ReadOnly Property QueryString() As String 'Implements BaseClasses.CPRequestBaseClass.QueryString
        Public MustOverride ReadOnly Property Referer() As String 'Implements BaseClasses.CPRequestBaseClass.Referer
        Public MustOverride ReadOnly Property RemoteIP() As String 'Implements BaseClasses.CPRequestBaseClass.RemoteIP
        Public MustOverride ReadOnly Property Secure() As Boolean 'Implements BaseClasses.CPRequestBaseClass.Secure
        Public MustOverride Function OK(ByVal RequestName As String) As Boolean 'Implements BaseClasses.CPRequestBaseClass.OK
    End Class

End Namespace

File: source/CPBase50/CPRequestBaseClass.cs
//========================================================================
// This conversion was produced by the Free Edition of
// Instant C# courtesy of Tangible Software Solutions.
// Order the Premium Edition at https://www.tangiblesoftwaresolutions.com
//========================================================================

//
// documentation should be in a new project that inherits these classes. The class names should be the object names in the actual cp project
//

//INSTANT C# NOTE: Formerly VB project-level imports:
using System;
using System.Collections;
using System.Collections.Generic;
using System.Diagnostics;

namespace Contensive.BaseClasses
{
	public abstract class CPRequestBaseClass
	{
		//Public Sub New(ByVal cmcObj As Contensive.Core.cpCoreClass, ByRef CPParent As CPBaseClass)
		public abstract string Browser {get;} //Implements BaseClasses.CPRequestBaseClass.Browser
		public abstract bool BrowserIsIE {get;} //Implements BaseClasses.CPRequestBaseClass.BrowserIsIE
		public abstract bool BrowserIsMac {get;} //Implements BaseClasses.CPRequestBaseClass.BrowserIsMac
		public abstract bool BrowserIsMobile {get;} //Implements BaseClasses.CPRequestBaseClass.BrowserIsMobile
		public abstract bool BrowserIsWindows {get;} //Implements BaseClasses.CPRequestBaseClass.BrowserIsWindows
		public abstract string BrowserVersion {get;} //Implements BaseClasses.CPRequestBaseClass.BrowserVersion
		public abstract string Cookie(string CookieName); //Implements BaseClasses.CPRequestBaseClass.Cookie
		public abstract string CookieString {get;} //Implements BaseClasses.CPRequestBaseClass.CookieString
		public abstract string Form {get;} //Implements BaseClasses.CPRequestBaseClass.Form
		public abstract string FormAction {get;} //Implements BaseClasses.CPRequestBaseClass.FormAction
		public abstract bool GetBoolean(string RequestName); //Implements BaseClasses.CPRequestBaseClass.GetBoolean
		public abstract DateTime GetDate(string RequestName); //Implements BaseClasses.CPRequestBaseClass.GetDate
		public abstract int GetInteger(string RequestName); //Implements BaseClasses.CPRequestBaseClass.GetInteger
		public abstract double GetNumber(string RequestName); //Implements BaseClasses.CPRequestBaseClass.GetNumber
		public abstract string GetText(string RequestName); //Implements BaseClasses.CPRequestBaseClass.GetText
		public abstract string Host {get;} //Implements BaseClasses.CPRequestBaseClass.Host
		public abstract string HTTPAccept {get;} //Implements BaseClasses.CPRequestBaseClass.HTTPAccept
		public abstract string HTTPAcceptCharset {get;} //Implements BaseClasses.CPRequestBaseClass.HTTPAcceptCharset
		public abstract string HTTPProfile {get;} //Implements BaseClasses.CPRequestBaseClass.HTTPProfile
		public abstract string HTTPXWapProfile {get;} //Implements BaseClasses.CPRequestBaseClass.HTTPXWapProfile
		public abstract string Language {get;} //Implements BaseClasses.CPRequestBaseClass.Language
		public abstract string Link {get;} //Implements BaseClasses.CPRequestBaseClass.Link
		public abstract string LinkForwardSource {get;} //Implements BaseClasses.CPRequestBaseClass.LinkForwardSource
		public abstract string LinkSource {get;} //Implements BaseClasses.CPRequestBaseClass.LinkSource
		public abstract string Page {get;} //Implements BaseClasses.CPRequestBaseClass.Page
		public abstract string Path {get;} //Implements BaseClasses.CPRequestBaseClass.Path
		public abstract string PathPage {get;} //Implements BaseClasses.CPRequestBaseClass.PathPage
		public abstract string Protocol {get;} //Implements BaseClasses.CPRequestBaseClass.Protocol
		public abstract string QueryString {get;} //Implements BaseClasses.CPRequestBaseClass.QueryString
		public abstract string Referer {get;} //Implements BaseClasses.CPRequestBaseClass.Referer
		public abstract string RemoteIP {get;} //Implements BaseClasses.CPRequestBaseClass.RemoteIP
		public abstract bool Secure {get;} //Implements BaseClasses.CPRequestBaseClass.Secure
		public abstract bool OK(string RequestName); //Implements BaseClasses.CPRequestBaseClass.OK
	}

}

